hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/ConsumerMessageSender.java

```diff
@@ -277,21 +290,24 @@ class ResponseHandlingListener implements java.util.function.Consumer<MessageSen
 
         private final Message message;
         private final HermesTimerContext timer;
+        private final ConsumerProfiler profiler;
 
-        public ResponseHandlingListener(Message message, HermesTimerContext timer) {
+        public ResponseHandlingListener(Message message, HermesTimerContext timer, ConsumerProfiler profiler) {
             this.message = message;
             this.timer = timer;
+            this.profiler = profiler;
         }
 
         @Override
         public void accept(MessageSendingResult result) {
             timer.close();
             loadRecorder.recordSingleOperation();
+            profiler.measure(Measurement.HANDLERS);
```

Comment:
Again: stopWatch stopped on a different thread than it was started on. Will this yield correct results?